library/src/main/java/ru/noties/markwon/spans/SpannableTheme.java

```diff
@@ -368,8 +378,16 @@ public int getCodeBackgroundColor(@NonNull Paint paint, boolean multiline) {
     }
 
     public void applyHeadingTextStyle(@NonNull Paint paint, @IntRange(from = 1, to = 6) int level) {
-        paint.setFakeBoldText(true);
-        paint.setTextSize(paint.getTextSize() * HEADING_SIZES[level - 1]);
+        if (headingTypeface == null) {
+            paint.setFakeBoldText(true);
+        } else {
+            paint.setTypeface(headingTypeface);
+        }
+        if (headingTextSizes != null && headingTextSizes.length >= level) {
```

Comment:
The check must be `headingTextSizes.length > level`, otherwise with `level=6` and `headingTextSizes.length = 6` we would get ArrayIndexOutOfBounds

Reply:
...I don't think this is correct.  If level = 6 and headingTextSizes.length = 6, the last index in the array will be 5.  when we index into it on the next line, we subtract one (6-1) = 5.  Unless I'm missing something?